src/shared/components/oncoprint/Oncoprint.tsx

```diff
@@ -108,16 +114,24 @@ interface IBaseHeatmapTrackSpec {
     data: IBaseHeatmapTrackDatum[];
     trackGroupIndex: number;
 }
-export interface IGeneHeatmapTrackSpec extends IBaseHeatmapTrackSpec {
-    data: IGeneHeatmapTrackDatum[];
-    onRemove: () => void;
+
+export interface IHeatmapTrackSpec extends IBaseHeatmapTrackSpec {
+    data: IBaseHeatmapTrackDatum[]; // can be IGeneHeatmapTrackDatum or ITreatmentHeatmapTrackDatum
     info?: string;
     labelColor?: string;
+    trackLinkUrl?: string | undefined;
+    onRemove: () => void;
+    molecularProfileName?: String
+    pivotThreshold?: number;
+    sortOrder?: SortOrder;
+    maxProfileValue?: number;
+    ruleSetTrackId?: number;
```

Comment:
sorry I just noticed this. but ruleSetTrackId doesn't belong here. Users of the oncoprint shouldn't have to specify that, it should be automatically handled. Maybe you are trying to spedcify wheher its a gene heatmap or a treatment heatmap? In that case please add an enum for that and have the user specify that, then we'll handle the ruleset sharing the same way we have been in existing code.

Reply:
I have removed the `ruleSetTrackId` member of the `IHeatmapTrackSpec` interface. I have replaced this with an extension of the `trackIdForRuleSetSharing` map in `transitionTracks()`:
```
    const trackIdForRuleSetSharing = {
        genetic: undefined as undefined|TrackId,
        genesetHeatmap: undefined as undefined|TrackId,
        heatmap: undefined as undefined|TrackId,
        heatmap01:undefined as undefined|TrackId,
        treatment: {} as any as TreatmentProfileToTrackIdMap  <----
    };
```
I think this solves your problem with attaching the ruleSetTrackId to Oncoprint.

Reply:
that's exactly what I was hoping for, thanks